src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/SingleThreadedDispatcher.cs

```diff
@@ -0,0 +1,32 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    internal abstract class SingleThreadedDispatcher
```

Comment:
Hmmm, while `SingleThreadedDispatcher` does in fact indicate what this class does I half wonder if we're being a litttle too ambiguous here. Aka, the purpose of this class is to have a single threaded operation that allows safe access to our `ProjectSnapshotManager`. If it isn't too much work (hopefully two find + replaces would be enough) would it be possible to rename this to something more like `ProjectSnapshotManagerDispatcher`? Everything else could stay the same because it'd all stem from a project snapshot manager dispatcher which would make sense as to how it should be used.

Reply:
Ahh that's a great point, I totally agree. I made the rename in my latest push 👍